controllers/azuremanagedcontrolplane_controller_test.go

```diff
@@ -176,3 +182,82 @@ func TestAzureManagedControlPlaneReconcilePaused(t *testing.T) {
 	g.Expect(err).To(BeNil())
 	g.Expect(result.RequeueAfter).To(BeZero())
 }
+
+func TestAzureManagedControlPlaneReconcileNormal(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	cp := &infrav1.AzureManagedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-azmp",
+			Namespace: "fake-ns",
+		},
+		Spec: infrav1.AzureManagedControlPlaneSpec{
+			AzureManagedControlPlaneClassSpec: infrav1.AzureManagedControlPlaneClassSpec{
+				Version: "0.0.1",
+			},
+		},
+		Status: infrav1.AzureManagedControlPlaneStatus{
+			Ready:       false,
+			Initialized: false,
+		},
+	}
+	scheme, err := newScheme()
+	g.Expect(err).ToNot(HaveOccurred())
+
+	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(cp).WithStatusSubresource(cp).Build()
+	amcpr := &AzureManagedControlPlaneReconciler{
+		Client: client,
+	}
+
+	helper, err := patch.NewHelper(cp, client)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	scopes := &scope.ManagedControlPlaneScope{
+		Cluster: &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "fake-cluster",
+				Namespace: "fake-ns",
+			},
+		},
+		Client:       client,
+		PatchHelper:  helper,
+		ControlPlane: cp,
+	}
+	scopes.SetAdminKubeconfigData(createFakeKubeConfig())
+	scopes.SetUserKubeconfigData(createFakeKubeConfig())
+
+	amcpr.getNewAzureManagedControlPlaneReconciler = func(scope *scope.ManagedControlPlaneScope) (*azureManagedControlPlaneService, error) {
+		ctrlr := gomock.NewController(t)
+		defer ctrlr.Finish()
+		svcr := mock_azure.NewMockServiceReconciler(ctrlr)
+		svcr.EXPECT().Reconcile(gomock.Any()).Return(nil).AnyTimes()
+
+		return &azureManagedControlPlaneService{
+			kubeclient: scopes.Client,
+			scope:      scopes,
```

Comment:
I know these have the same value, but would it technically be more correct to refer to the `scope` parameter instead of `scopes` here? Otherwise I see `scope` isn't used.

Reply:
I'm not following exactly what you are referring to.  The scopes needs to carry the client, controlplane, etc.  Are you referring to the kubeclient? which is a client.Client?

Reply:
I mean the `scope` parameter to this function instead of the `scopes` variable defined outside this function.

Reply:
Yeah that makes sense, and would much rather have it reflect the value passed in (which is scopes) but get it's value from the param in the func we are patching.  

Nice catch.  Thanks for that @nojnhuh

Reply:
Updated and pushed.  

Now we should be good to go @nojnhuh